bundles/org.openhab.binding.enocean/src/main/java/org/openhab/binding/enocean/internal/eep/EEPHelper.java

```diff
@@ -60,4 +64,19 @@ public static State validateTotalUsage(State value, State currentState, Configur
 
         return value;
     }
+
+    public static double calculateScaledValue(int unscaledValue, double scaledMin, double scaledMax, double unscaledMin,
+            double unscaledMax) {
+
+        // Validation checkup
+        if (unscaledValue < unscaledMin) {
+            logger.warn("Unscaled value ({}) lower than the minimum allowed ({})", unscaledValue, unscaledMin);
+            return 0;
```

Comment:
@jlaur this is something unexpected.
I am trying to handle the case when a device will send message with value which is not in the unscaled range ( see [A5_09_04](http://tools.enocean-alliance.org/EEPViewer/profiles/A5/09/04/A5-09-04.pdf) documentation for humidity - the humidity is frofm 0 to 8 - which means that there could be a value from 0 to 255 but unscaled range is from 0 to 200.
I have seen this more than once and I belive if a device send such a value it is not in the standard.

I don't have strong opinion how this should be handled and I am open to suggestions.

I think till now this case is not handled at all, there is no validation for the expected values according to the standard.

Reply:
If this can not be represented as a valid number in such cases, from a user perspective this should probably result in state `UNDEF`, and there should be DEBUG or TRACE logging.

Implementation-wise, I would suggest two things:
- Guard this method by throwing `IllegalArgumentException`.
- Avoid calling this method in such cases, but log and set state `UNDEF`.

Reply:
@jlaur What would you say to replace the `calculateScaledValue` function with this one:
```
    public static boolean validateUnscaledValue(int unscaledValue, double unscaledMin, double unscaledMax) {                                 
        if (unscaledValue < unscaledMin) {                                                                                                   
            logger.debug("Unscaled value ({}) lower than the minimum allowed ({})", unscaledValue, unscaledMin);
            return false;
        } else if (unscaledValue > unscaledMax) {
            logger.debug("Unscaled value ({}) bigger than the maximum allowed ({})", unscaledValue, unscaledMax);
            return false;
        }

        return true;
    }

    public static State calculateState(int unscaledValue, double scaledMin, double scaledMax, double unscaledMin,
            double unscaledMax, Unit<?> unit) {

        if (!validateUnscaledValue(unscaledValue, unscaledMin, unscaledMax)) {
            return UnDefType.UNDEF;
        }

        double scaledValue = scaledMin + ((unscaledValue * (scaledMax - scaledMin)) / (unscaledMax - unscaledMin));
        return new QuantityType<>(scaledValue, unit);
    }
```
Or with throwing exceptions like this
```
    public static void validateUnscaledValue(int unscaledValue, double unscaledMin, double unscaledMax)
        throws IllegalArgumentException {
        
        if (unscaledValue < unscaledMin) {
            throw new IllegalArgumentException(
                    String.format("Unscaled value (%.2%f) lower than the minimum allowed (%.2%f)",
                        unscaledValue, unscaledMin));
        } else if (unscaledValue > unscaledMax) {
            throw new IllegalArgumentException( 
                    String.format("Unscaled value (%.2%f) bigger than the maximum allowed (%.2%f)",
                        unscaledValue, unscaledMax));
        }               
    }   
    
    public static State calculateState(int unscaledValue, double scaledMin, double scaledMax, double unscaledMin,
            double unscaledMax, Unit<?> unit) {
            
        try {
            validateUnscaledValue(unscaledValue, unscaledMin, unscaledMax);
        } catch (IllegalArgumentException e) {
            return UnDefType.UNDEF;
        }   

        double scaledValue = scaledMin + ((unscaledValue * (scaledMax - scaledMin)) / (unscaledMax - unscaledMin));
        return new QuantityType<>(scaledValue, unit);
    }

Reply:
It's up to you. 🙂 Personally I would go with the non-throwing version since it's faster. Usually `IllegalArgumentException` is used to check pre-conditions in order to expose bugs, so it's not something you would usually catch, but should rather prevent calling the method with invalid arguments.